src/components/Button.js

```diff
@@ -33,19 +33,38 @@ const getColor = (props, key, shade) => {
   }
 }
 
+const squareWidth = val => val / 9 * 10
+
 const minWidth = props => {
   if (props.w == null && !props.square) {
-    return 135
+    // return 135
+    return 'auto'
+  }
+  if (props.square) {
+    return props.emphasized
+      ? squareWidth(props.theme.space[5])
+      : squareWidth(props.theme.space[4])
+  }
+
+  return props.w
+}
+
+const horizontalPadding = props => {
+  if (props.w == null && !props.square) {
+    // return 20
+    return props.emphasized ? 50 : 20
   }
   if (props.square) {
-    return props.emphasized ? props.theme.space[5] : props.theme.space[4]
+    return 0
   }
 
   return props.w
```

Comment:
This line is causing issues for fixed width buttons, as it sets the padding to the width of the button! I think this could be safely changed to:
```
if (props.square || props.w) {
   return 0
}
return props.emphasized ? 50 : 20
```